src/whitenoise/app_update.rs

```diff
@@ -0,0 +1,531 @@
+use nostr_sdk::prelude::*;
+use std::str::FromStr;
+use std::time::Duration;
+
+use crate::whitenoise::error::{Result, WhitenoiseError};
+
+const ZAPSTORE_RELAY_URL: &str = "wss://relay.zapstore.dev";
+const WHITE_NOISE_PUBKEY: &str = "75d737c3472471029c44876b330d2284288a42779b591a2ed4daa1c6c07efaf7";
+const TIMEOUT: Duration = Duration::from_secs(10);
+
+#[derive(Debug, Clone)]
+pub struct AppUpdateInfo {
+    pub version: String,
+    pub update_available: bool,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+struct Version {
+    major: u32,
+    minor: u32,
+    patch: u32,
+}
```

Comment:
Version only handles strict `x.y.z` format, let's make sure / validate maybe with Jeff that we don't need to support things like:

- Pre-release: `1.0.0-alpha`, `1.0.0-rc.1`
- Build metadata: `1.0.0+build.123`

There are some options here, ranging from fully supporting semver to leaving it as is and committing to strict `x.y.z` version format. Perhaps just stripping suffixes could be enough.

Reply:
I'm with you on doing full semver. However the goal right now is to make this work with what we have at the moment so I'll add this to the discussion on the issue as part of the future updates to this feature.